### PR TITLE
DOC: update NEP29 to address PEP0602

### PR DESCRIPTION
**Pull request by tacaswell**
Following up on the discussion on the mailing list, update the prose of NEP29 to account for PEP0602.

attn @asmeurer

### CODE REVIEW

**Review by charris — APPROVED** [review]

File: doc/neps/nep-0029-deprecation_policy.rst [hunk 1]
```diff
@@ -74,11 +74,19 @@ a project with a major or minor version released in December 2019 should
...
+Python in the window.  The window was extended 6 months beyond the anticipated
+two-release interval for Python to provide resilience against small
+fluctuations / delays in its release schedule.
+
+The Python release cadence was increased to 12 months in `PEP 0602
+<https://peps.python.org/pep-0602/>`__ so there will now be 3-4 Python releases
+in the support window at any time.  However, PEP 0602 does not decrease the
+support window of Python (18 months of regular full bug-fix releases and 42
+months of as-needed source-only releases).  Thus, we do not expect an increase
+in which our users upgrade their Python and our 42 month support window will
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply doc/neps/nep-0029-deprecation_policy.rst]: Yes, that is the intent, thank you @mattip !

**Comment by tacaswell** [comment]
I'm a bit worried this did not get enough buy-in before merging, but I suppose it can be further edited if there are major protests to this wording.

**Comment by seberg** [comment]
Maybe repeat the change on the mailing list?  My current understanding is that it de-facto just clarifies the intention, so missing buy-in may be more of "lets change it"?
(The possible move to 1 year was anticipated at the time after all.)

File: doc/neps/nep-0029-deprecation_policy.rst [hunk 2]
```diff
@@ -74,11 +74,19 @@ a project with a major or minor version released in December 2019 should
...
-in the window.  The window is extended 6 months beyond the anticipated two-release
-interval for Python to provide resilience against small fluctuations /
-delays in its release schedule.
+When this NEP was drafted the Python release cadence was 18 months so a 42
+month window ensured that there would always be at least two minor versions of
+Python in the window.  The window was extended 6 months beyond the anticipated
+two-release interval for Python to provide resilience against small
+fluctuations / delays in its release schedule.
+
+The Python release cadence was increased to 12 months in `PEP 0602
```
Comment (stefanv) [inline doc/neps/nep-0029-deprecation_policy.rst:83]: This is a bit nitpicky, but I'd say cadence is frequency, whereas you are referring to interval (/period) here.  So, I'd have used `The Python release interval was decreased to 12 months"

**Review by stefanv — COMMENTED** [review]
This looks good to me; thanks @tacaswell

**Comment by rgommers** [comment]
LGTM too, thanks @tacaswell

File: doc/neps/nep-0029-deprecation_policy.rst (hunk 2 above)
Comment (tacaswell) [reply doc/neps/nep-0029-deprecation_policy.rst:83]: Yeah, writing "increased to a --smaller number-- " felt a bit weird..

How about: "The Ptyhon release cadenced increased and releases are now every 12 months pre `PEP...`"

Comment (stefanv) [reply doc/neps/nep-0029-deprecation_policy.rst:83]: "increased, with releases now every 12 months" maybe? Anyway 👍

**Comment by tacaswell** [comment]
Created https://github.com/numpy/numpy/pull/21616 with @stefanv 's suggested edit.

**Pull request merged**